### PR TITLE
[DebugInfo][RemoveDIs] Add flag to use "new" debug-info  in opt

### PR DESCRIPTION
**Pull request by jmorse**
Our option to turn on the non-intrinsic form of debug-info (`--experimental-debuginfo-iterators`) currently requires that LLVM is built with the `LLVM_EXPERIMENTAL_DEBUGINFO_ITERATORS` cmake flag enabled, so that some (slight) performance regressions aren't on-by-default during the prototype/testing period. However, we still want to be able to _optionally_ run tests, if support is built into LLVM.

To allow optionally exercising the non-intrinsic debug-info code, this patch adds `--try-experimental-debuginfo-iterators` to opt, which turns the `--experimental-debuginfo-iterators` flag on if support is built in, or leaves it off. This means we can run tests that:
 * Use normal dbg.value intrinsics if there's no support, or
 * Uses non-instruction DPValues if there is support.
  
Which means we can start getting test coverage of DPValues/RemoveDIs behaviour, from in-tree tests, on our RemoveDIs buildbot. All the code to do with automagically converting from one form to the other landed in 10a9e7442c4c4e.

### CODE REVIEW

File: llvm/tools/opt/opt.cpp [hunk 1]
```diff
@@ -438,6 +444,19 @@ int main(int argc, char **argv) {
...
+  if (TryUseNewDbgInfoFormat) {
+#ifdef EXPERIMENTAL_DEBUGINFO_ITERATORS
+    // If LLVM was built with support for this, turn the new debug-info format
+    // on.
+    UseNewDbgInfoFormat = true;
+#else
+    // It it wasn't, do nothing.
+    ;
+#endif
+  }
```
Comment (OCHyams) [inline llvm/tools/opt/opt.cpp:458]:
```suggestion
  // RemoveDIs debug-info transition: tests may request that we /try/ to use the
  // new debug-info format, if it's built in.
#ifdef EXPERIMENTAL_DEBUGINFO_ITERATORS
  if (TryUseNewDbgInfoFormat)
    UseNewDbgInfoFormat = true;
#endif
  }
```

Comment (OCHyams) [reply llvm/tools/opt/opt.cpp:458]: Unless you wanted to add prints etc to the false branch, I don't think we need it?

**Review by OCHyams — APPROVED** [review]
SGTM

File: llvm/tools/opt/opt.cpp (hunk 1 above)
Comment (pogo59) [reply llvm/tools/opt/opt.cpp:458]: @OCHyams might need a `(void)TryUseNewDbgInfoFormat;` to suppress an unused-variable warning in the iterators-off mode.

**Pull request merged**